include/zephyr/bluetooth/audio/audio.h

```diff
@@ -32,93 +32,26 @@ extern "C" {
- *  music and in-game voice chat between participants; or a mix of all the above.
- */
-#define BT_AUDIO_CONTEXT_TYPE_GAME               BIT(3)
-/** @def BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL
- *  @brief Instructional audio
- *
- *  Instructional audio as, for example, in navigation, traffic announcements or user guidance.
- */
-#define BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL      BIT(4)
-/** @def BT_AUDIO_CONTEXT_TYPE_VOICE_ASSISTANTS
- *  @brief Voice assistant audio
- *
- *  Man-machine communication, for example, with voice recognition or virtual assistants.
- */
-#define BT_AUDIO_CONTEXT_TYPE_VOICE_ASSISTANTS   BIT(5)
-/** @def BT_AUDIO_CONTEXT_TYPE_LIVE
- *  @brief Live audio
- *
- *  Live audio, for example, from a microphone where audio is perceived both through
- *  a direct acoustic path and through an LE Audio Stream.
- */
-#define BT_AUDIO_CONTEXT_TYPE_LIVE               BIT(6)
-/** @def BT_AUDIO_CONTEXT_TYPE_SOUND_EFFECTS
- *  @brief Sound effects
- *
- *  Sound effects including keyboard and touch feedback; menu and user interface sounds;
- *  and other system sounds.
- */
-#define BT_AUDIO_CONTEXT_TYPE_SOUND_EFFECTS      BIT(7)
-/** @def BT_AUDIO_CONTEXT_TYPE_NOTIFICATIONS
- *  @brief Notification and reminder sounds
- *
- *  Notification and reminder sounds; attention-seeking audio, for example, in beeps signaling
- *  the arrival of a message.
- */
-#define BT_AUDIO_CONTEXT_TYPE_NOTIFICATIONS      BIT(8)
-/** @def BT_AUDIO_CONTEXT_TYPE_RINGTONE
- *  @brief Ringtone as in a call alert
- *
- * Alerts the user to an incoming call, for example, an incoming telephony or video call,
- * including traditional cellular as well as VoIP and Push-to-Talk.
- */
-#define BT_AUDIO_CONTEXT_TYPE_RINGTONE           BIT(9)
-/** @def BT_AUDIO_CONTEXT_TYPE_ALERTS
- *  @brief Alarms and timers
- *
- *  Alarms and timers; immediate alerts, for example, in a critical battery alarm, timer expiry or
- *  alarm clock, toaster cooker, kettle, microwave, etc.
- */
-#define BT_AUDIO_CONTEXT_TYPE_ALERTS             BIT(10)
-/** @def BT_AUDIO_CONTEXT_TYPE_EMERGENCY_ALARM
- *  @brief Emergency alarm
- *
- *  Emergency alerts as, for example, with fire alarms or other urgent alerts.
+ * These values are defined by the Generic Audio Assigned Numbers, bluetooth.com
  */
-#define BT_AUDIO_CONTEXT_TYPE_EMERGENCY_ALARM    BIT(11)
+enum bt_audio_context {
+	BT_AUDIO_CONTEXT_TYPE_PROHIBITED = 0,
+	BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED = BIT(0),
+	BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL = BIT(1),
+	BT_AUDIO_CONTEXT_TYPE_MEDIA = BIT(2),
+	BT_AUDIO_CONTEXT_TYPE_GAME = BIT(3),
+	BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL = BIT(4),
+	BT_AUDIO_CONTEXT_TYPE_VOICE_ASSISTANTS = BIT(5),
+	BT_AUDIO_CONTEXT_TYPE_LIVE = BIT(6),
+	BT_AUDIO_CONTEXT_TYPE_SOUND_EFFECTS = BIT(7),
+	BT_AUDIO_CONTEXT_TYPE_NOTIFICATIONS = BIT(8),
+	BT_AUDIO_CONTEXT_TYPE_RINGTONE = BIT(9),
+	BT_AUDIO_CONTEXT_TYPE_ALERTS = BIT(10),
+	BT_AUDIO_CONTEXT_TYPE_EMERGENCY_ALARM = BIT(11),
+};
```

Comment:
Good that it is now an enum :) 

Any reason why you omitted the documentation for each type though?

Reply:
The names are intuitive and are well documented in assigned numbers

Reply:
If we rely on the assigned number documentation, then we should reference that.

Not sure if I agree that the names are enough, as something like `BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL` may not be clear, and that it perhaps also isn't clear what the difference between `BT_AUDIO_CONTEXT_TYPE_NOTIFICATIONS` and `BT_AUDIO_CONTEXT_TYPE_SOUND_EFFECTS` is, and how `BT_AUDIO_CONTEXT_TYPE_ALERTS` and `BT_AUDIO_CONTEXT_TYPE_EMERGENCY_ALARM` differ from each other. 

So I suggest to either reference the Generic Audio assigned numbers, or that we keep the documentation.

Reply:
> If we rely on the assigned number documentation, then we should reference that.
> 
> Not sure if I agree that the names are enough, as something like `BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL` may not be clear, and that it perhaps also isn't clear what the difference between `BT_AUDIO_CONTEXT_TYPE_NOTIFICATIONS` and `BT_AUDIO_CONTEXT_TYPE_SOUND_EFFECTS` is, and how `BT_AUDIO_CONTEXT_TYPE_ALERTS` and `BT_AUDIO_CONTEXT_TYPE_EMERGENCY_ALARM` differ from each other.
> 
> So I suggest to either reference the Generic Audio assigned numbers, or that we keep the documentation.

```
/** @brief Audio Context Type for Generic Audio
 *
 * These values are defined by the Generic Audio Assigned Numbers, bluetooth.com
 */
enum bt_audio_context {
```
is already there. Is that not enough?

Reply:
Oh, sorry, must've missed that :)